src/Compilers/CSharp/Portable/Emitter/Model/NamedTypeSymbolAdapter.cs

```diff
@@ -797,7 +797,7 @@ string Cci.INamedEntity.Name
             {
                 if (AdaptedNamedTypeSymbol.IsExtension)
                 {
-                    throw ExceptionUtilities.Unreachable();
+                    return AdaptedNamedTypeSymbol.ExtensionGroupingName;
```

Comment:
This is not the right thing to do. The exception is intentional because the symbol is a source artifact, not a metadata artifact. Therefore, it should never get accessed as one during emit.